### PR TITLE
tt-FLux, tt_dit refactor

### PR DESCRIPTION
**Pull request by sadesoyeTT**
### Ticket
N/A

### What's changed
Add Flux.1 model. Reorganize tt-dit

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/18851055468) CI passes

Tests

- [x] [T3K Frequent](https://github.com/tenstorrent/tt-metal/actions/runs/18825871908/job/53708697349)

- [x] [T3K Perf](https://github.com/tenstorrent/tt-metal/actions/runs/18788268463/job/53612773927)

- [x] [T3K demo](https://github.com/tenstorrent/tt-metal/actions/runs/18822147316/job/53699492419)

- [x] [Galaxy Frequent](https://github.com/tenstorrent/tt-metal/actions/runs/18797611287/job/53640541752)

- [x] [Galaxy Demo](https://github.com/tenstorrent/tt-metal/actions/runs/18825772639/job/53708554502)

- [x] [Galaxy Perf](https://github.com/tenstorrent/tt-metal/actions/runs/18852936550/job/53796450732). Bump perf target

### CODE REVIEW

**Review by mbahnasTT — APPROVED** [review]

File: models/experimental/tt_dit/blocks/attention.py [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+        padding_len = cls.spatial_sequence_padding_length(length=x.shape[-2], sp_factor=sp_factor)
+        return torch.nn.functional.pad(x, (0, 0, 0, padding_len))
+
+
+def _apply_rope(x: ttnn.Tensor, freqs_cis: tuple[ttnn.Tensor, ttnn.Tensor]) -> ttnn.Tensor:
+    cos, sin = freqs_cis
+    cos = cos.reshape([1, 1, *cos.shape])
+    sin = sin.reshape([1, 1, *sin.shape])
+
+    return x * cos + ttnn.alt_complex_rotate90(x) * sin
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/blocks/attention.py:346]: how does this _apply_rope perf compare to the previous usage of the fused ROPE op?

File: models/experimental/tt_dit/blocks/attention.py [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+
+
+# TODO: move to module.py
+class UnregisteredModule:
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/blocks/attention.py]: Does this need to be here?

File: models/experimental/tt_dit/layers/embeddings.py [hunk 3]
```diff
@@ -83,6 +83,109 @@ def forward(self, timestep: ttnn.Tensor, pooled_projection: ttnn.Tensor) -> ttnn
...
+
+        self.timestep_embedder = TimestepEmbedding(256, embedding_dim, mesh_device=mesh_device)
+        self.guidance_embedder = (
+            TimestepEmbedding(256, embedding_dim, mesh_device=mesh_device) if with_guidance else None
+        )
+        self.text_embedder = PixartAlphaTextProjection(pooled_projection_dim, embedding_dim, mesh_device=mesh_device)
+
+        self.time_proj_factor = self._create_time_proj_factor(256)
+
+    # def to_cached_state_dict(self, path_prefix):
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/layers/embeddings.py]: delete commented code

File: models/experimental/tt_dit/models/transformers/transformer_flux1.py [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+        # combined_normed = ttnn.squeeze(self.norm(ttnn.unsqueeze(combined, 0)), 0)
+        spatial_normed = ttnn.squeeze(self.norm(ttnn.unsqueeze(spatial, 0)), 0)
+        prompt_normed = ttnn.squeeze(self.norm(ttnn.unsqueeze(prompt, 0)), 0)
+
+        shift_msa, scale_msa, gate_msa = _chunk_time3d(time, 3)
+        # norm_combined = combined_normed * (1 + scale_msa) + shift_msa
+        norm_spatial = spatial_normed * (1 + scale_msa) + shift_msa
+        norm_prompt = prompt_normed * (1 + scale_msa) + shift_msa
+
+        # norm_combined = self.ccl_manager.all_gather_persistent_buffer(
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/models/transformers/transformer_flux1.py]: delete commented code

File: models/experimental/tt_dit/models/transformers/transformer_flux1.py [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+                spatial=spatial,
+                prompt=prompt,
+                time_embed=time_embed,
+                spatial_rope=spatial_rope,
+                prompt_rope=prompt_rope,
+                spatial_sequence_length=spatial_sequence_length,
+                skip_time_embed_activation_fn=True,
+            )
+
+            if i % 6 == 0:
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/models/transformers/transformer_flux1.py]: should this be removed?

File: models/experimental/tt_dit/models/transformers/transformer_flux1.py [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+                ttnn.ReadDeviceProfiler(spatial.device())
+
+        prompt = ttnn.clone(prompt, dtype=spatial.dtype)
+
+        # combined = ttnn.concat([prompt, spatial], dim=1)
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/models/transformers/transformer_flux1.py]: delete commented code

File: models/experimental/tt_dit/models/transformers/transformer_flux1.py [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+        [scale, shift] = _chunk_time3d(spatial_time, 2)
+
+        spatial = self.ccl_manager.all_gather_persistent_buffer(spatial, dim=2, mesh_axis=tp_axis, use_hyperparams=True)
+
+        spatial = spatial * (1 + scale) + shift
+
+        return self.proj_out(spatial)
+
+
+def _chunk_time3d(t: ttnn.Tensor, count: int) -> list[ttnn.Tensor]:
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/models/transformers/transformer_flux1.py]: duplicate impl of chunk_time3d?

File: models/experimental/tt_dit/models/transformers/wan2_2/attention_wan.py [hunk 8]
```diff
@@ -268,7 +268,7 @@ def create_heads(inp):
                     num_links=self.ccl_manager.num_links,
                     cluster_axis=self.parallel_config.sequence_parallel.mesh_axis,
                     mesh_device=self.mesh_device,
-                    topology=ttnn.Topology.Linear,  # RJA always uses Linear topology
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/models/transformers/wan2_2/attention_wan.py:271]: this is probably a merge mistake - please revert this change. RJA will hang if topology is ring, and Wan now uses ring topology on 6U.

File: models/experimental/tt_dit/models/vae/vae_sd35.py [hunk 9]
```diff
@@ -57,6 +57,7 @@ def __init__(
             kernel_size=(3, 3),
             mesh_device=mesh_device,
             mesh_axis=parallel_config.tensor_parallel.mesh_axis,
+            sp_axis=parallel_config.tensor_parallel.mesh_axis,
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/models/vae/vae_sd35.py:60]: I think I don't understand this bit - why is sp_axis same as mesh_axis?

File: models/experimental/tt_dit/tests/models/wan2_2/test_attention_wan.py [hunk 10]
```diff
@@ -15,12 +15,11 @@
 from ....parallel.config import DiTParallelConfig, ParallelFactor
 from ....utils.padding import pad_vision_seq_parallel
 from ....utils.mochi import get_rot_transformation_mat, stack_cos_sin
-from ....utils.test import ring_params, line_params
 from diffusers import WanTransformer3DModel
```
Comment (cglagovichTT) [inline models/experimental/tt_dit/tests/models/wan2_2/test_attention_wan.py:21]: please revert this change

Comment (cglagovichTT) [reply models/experimental/tt_dit/tests/models/wan2_2/test_attention_wan.py:21]: and all other changes to Wan regarding topology

File: models/experimental/tt_dit/models/vae/vae_sd35.py (hunk 9 above)
Comment (sadesoyeTT) [reply models/experimental/tt_dit/models/vae/vae_sd35.py:60]: Essentially, some non SP ops need their inputs to be gathered. sp_axis helps keep track on what axis to gather on. Quite overloaded. This is more of a design and constraint issue. Because of GN that supports SP, I delay ccl calls to the op that requires it. That decision works well for the VAE because there are no branching (apart from Resnet which is not an issue). However, the implementation breaks in the usecase above. I plan to refactor and have some update at the next run with motif. I didn't want to delay flux further, so I did the quickest thing.

**Review by williamlyTT — APPROVED** [review]
